packages/conferer/src/Conferer/FromConfig/Internal.hs

```diff
@@ -152,15 +153,16 @@ instance FromConfig LBS.ByteString where
 
 instance forall a. (Typeable a, FromConfig a) => FromConfig (Maybe a) where
   fetchFromConfig key config = do
-    let
-      newConfig =
-        case getKeyFromDefaults key config >>= fromDynamic @(Maybe a) of
+
+    newConfig <-
+        fetchFromDefaults @(Maybe a) key config >>=
+        \case
         Just (Just defaultThing) -> do
-          config & addDefault key defaultThing
+          return $ config & addDefault key defaultThing
         Just Nothing -> do
-          config & removeDefault key
+          return config
         _ -> do
-          config
+          return config
```

Comment:
💭 Do we need the `Just Nothing` case?, that one and the next are the same

Also, it was a bit hard for me to follow this code 😅 . Let me see if I got it right:

If there's a default value that can be parsed to the expected type then we add it to the config and then (after this lines) we use the usual fetchFromConfig logic over that new config, right?
If that's the case, can we do it like this?:

```haskell
let possibleDefaultValue = join $ fetchFromDefaults @(Maybe a) key config
    newConfig = case possibleDefaultValue of
        Just defaultValue -> addDefault key defaultValue config
        Nothing -> config
```


Also, why are we not removing it anymore? 🤔 , what does that mean?

Reply:
Those states mean:
* `Nothing`: There is no default
* `Just Nothing`: There is a default and its value is `Nothing`
* Just (Just something): This is a default and its value is `Just something`

The point for this is that you can specify the default as a `Maybe a` instead of an `a`.

Previously the second case created problems with the `FromConfig` `a` instance, since it would always trigger a wrong type error, now that we don't have those we can avoid doing that.

Also since it doesn't throw anymore I can remove the need for the `>>=` but I wouldn't use the `join` since the different maybes mean different things

Reply:
> Those states mean:
>
>    Nothing: There is no default
>    Just Nothing: There is a default and its value is Nothing
>    Just (Just something): This is a default and its value is Just something

In that case, why are we handling `Just Nothing` and `Nothing` the same way?

Why not do something like:
```
Just maybeValue -> addDefault key defaultValue config # the value being added here is a Maybe
Nothing -> config
```
?